### PR TITLE
makefile: klayout.lyt works from outside ORFS folder

### PR DESCRIPTION
**Pull request by oharboe**
If klayout 0.28.11 is available, use feature to expand FLOW_HOME environment variable to find PDK files.

https://github.com/KLayout/klayout/commit/f518d1aa1d974897cfeb1ef057057125eeceef40

By using this feature if it is available, the feature can be deployed now and make the road a bit bump across deployment scenarios. Especially considering that klayout is a slightly more peripheral dependency where it is a bit tougher to require an exact version.

Later on when 0.28.11 is everywhere, the cruft can be cleaned up.

### CODE REVIEW

**Comment by vijayank88** [comment]
@oharboe 
CI failures with file path
```
cp platforms/nangate45/lef/NangateOpenCellLibrary.tech.lef objects/nangate45/gcd/base/klayout_tech.lef
SC_LEF_RELATIVE_PATH="$\(env('FLOW_HOME')\)/platforms/nangate45/lef/NangateOpenCellLibrary.macro.mod.lef"; \
OTHER_LEFS_RELATIVE_PATHS=$(echo "<lef-files>$(realpath --relative-to=./results/nangate45/gcd/base ./objects/nangate45/gcd/base/klayout_tech.lef)</lef-files>"); \
sed 's,<lef-files>.*</lef-files>,<lef-files>'"$SC_LEF_RELATIVE_PATH"'</lef-files>'"$OTHER_LEFS_RELATIVE_PATHS"',g' platforms/nangate45/FreePDK45.lyt > objects/nangate45/gcd/base/klayout.lyt
(/usr/bin/time -f 'Elapsed time: %E[h:]min:sec. CPU time: user %U sys %S (%P). Peak memory: %MKB.' stdbuf -o L /usr/bin/klayout -zz -rd design_name=gcd \
        -rd in_def=results/nangate45/gcd/base/6_final.def \
        -rd in_files="./platforms/nangate45/gds/NangateOpenCellLibrary.gds  " \
        -rd config_file= \
        -rd seal_file="" \
        -rd out_file=results/nangate45/gcd/base/6_1_merged.gds \
        -rd tech_file=./objects/nangate45/gcd/base/klayout.lyt \
        -rd layer_map= \
        -r ./util/def2stream.py) 2>&1 | tee ./logs/nangate45/gcd/base/6_1_merge.log
[INFO] Reporting cells prior to loading DEF ...
[INFO] Reading DEF ...
ERROR: Unable to open file: /OpenROAD-flow-scripts/flow/results/nangate45/gcd/base//$(env('FLOW_HOME'))/platforms/nangate45/lef/NangateOpenCellLibrary.macro.mod.lef (errno=2) in Layout.read
  ./util/def2stream.py:113
Command exited with non-zero status 1
```

**Comment by oharboe** [comment]
@vijayank88 Thanks! Hopefully I got it right this time....

**Comment by oharboe** [comment]
@maliberty unrelated CI failures

**Comment by maliberty** [comment]
How does this relate to #1439 ?

**Comment by oharboe** [comment]
> How does this relate to #1439 ?

Completely orthogonal.

File: flow/Makefile [hunk 1]
```diff
@@ -129,7 +129,7 @@ SHELL          := /usr/bin/env bash
 # - the following settings allowed user to point OpenROAD binaries to different
 #   location
 # - default is current install / clone directory
-export FLOW_HOME ?= .
+export FLOW_HOME ?= $(shell pwd)
```
Comment (maliberty) [inline flow/Makefile:132]: Does this cause 'make issue..' to use absolute paths rather than relative ones?  That would be undesirable.

Comment (oharboe) [reply flow/Makefile:132]: Good catch in terms of thinking of edge cases. I didn't think to test it before you mentioned it.

Works fine though:

```
$ make floorplan_issue
$ grep /home  vars-gcd-nangate45-base.sh  |wc -l
0
$ head -n 10 vars-gcd-nangate45-base.sh 
export FLOW_HOME=${FLOW_HOME:-$(pwd)}
export RCX_RULES="${FLOW_HOME}/platforms/nangate45/rcx_patterns.rules"
export DESIGN_NAME="gcd"
export SDC_FILE="./designs/nangate45/gcd/constraint.sdc"
export IO_PLACER_H="metal5"
export GDS_FILES="${FLOW_HOME}/platforms/nangate45/gds/NangateOpenCellLibrary.gds"
export CTS_BUF_CELL="BUF_X4"
export STREAM_SYSTEM="GDS"
export SYNTH_HIERARCHICAL="0"
export LATCH_MAP_FILE="${FLOW_HOME}/platforms/nangate45/cells_latch.v"
```

**Review by maliberty — APPROVED** [review]
